### ISSUE TITLE
path.getabsolute() invalidates server paths

### ISSUE THREAD

**Issue by Gaztin** [premake/premake-core#1129]
At work, we keep our third parties on our storage server. The paths looks something like this:

`//server/libraries/...`

But when these paths are passed through path.getabsolute() it shaves off the first parenthesis. Until now we've stuck with a workaround, but I'd like to fix the problem at its source.

I've debugged the program and found the cultprit:
https://github.com/premake/premake-core/blob/5dfb0238bc309df04819dd430def621ce854678d/src/host/path_getabsolute.c#L33-L37

Before I take a stab at it, can anyone else think of a good solution off the top of their head?

Edit: This problem occurs in path.getrelative() as well.

**Comment by Gaztin** [comment]
This seems to work. All existing tests ran successfully too. I'll open a PR later today.

```lua
result[0] = '\0';
int slash_pos = 0;
while (buffer[slash_pos++] == '/') {
	strcat(result, "/");
}
```

**Comment by tdesveauxPKFX** [comment]
This looks like a normalization path while appending the path.
Your solution would result in a path starting with multiple separator to not be properly normalized.

Example: `////dir/file` would keep all the starting `/` where with the current implementation it would result in `/dir/file`.

I think a server path is as much a special case as absolute path on windows, so I would do something like:
```c
result[0] = '\0';
if (buffer[0] == '/') {
	strcat(result, "/");

	if (buffer[1] == '/')
		strcat(result, "/");
}
```

Ultimately, I think getabsolute would deserve a refactor that concatenate the paths then call normalize on it.

**Comment by Gaztin** [comment]
Yes, you're right. a while-loop is not the correct solution.

I did some digging and found this on the [Open Group specification for POSIX 2017](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_271):
*(from term 3.271: Pathname)*
> Multiple successive \<slash\> characters are considered to be the same as one \<slash\>, except for the case of exactly two leading \<slash\> characters.

Which means it's not just a Windows thing like I thought.

**Comment by samsinsane** [comment]
I believe this is now resolved by #1136, if not please feel free to reopen!
